### PR TITLE
Fixed issues with  json key in trivy operator 

### PR DESCRIPTION
**Pull request by btwseeu78**
## Related issues
- https://github.com/aquasecurity/trivy-operator/issues/1298

## Checklist
- [x] I've read the [guidelines for contributing](https://github.com/aquasecurity/trivy-operator/blob/main/CONTRIBUTING.md) to this repository.
- [ ] I've added tests that prove my fix is effective or that my feature works.
- [ ] I've updated the [documentation](https://github.com/aquasecurity/trivy-operator/tree/main/docs) with the relevant information (if needed).
- [ ] I've added usage information (if the PR introduces new options)
- [ ] I've included a "before" and "after" example to the description (if the PR is a user interface change).

### CODE REVIEW

File: pkg/plugins/trivy/plugin.go [hunk 1]
```diff
@@ -801,32 +802,64 @@ func (p *plugin) getPodSpecForStandaloneMode(ctx trivyoperator.PluginContext, co
...
 			registryUsernameKey := fmt.Sprintf("%s.username", c.Name)
 			registryPasswordKey := fmt.Sprintf("%s.password", c.Name)
-
-			env = append(env, corev1.EnvVar{
-				Name: "TRIVY_USERNAME",
-				ValueFrom: &corev1.EnvVarSource{
-					SecretKeyRef: &corev1.SecretKeySelector{
-						LocalObjectReference: corev1.LocalObjectReference{
-							Name: secret.Name,
+			if gcrImage {
```
Comment (chen-keinan) [inline pkg/plugins/trivy/plugin.go:809]: need also to make sure that secret password is from type (sa) :
```
{
  "type": "service_account",
  "project_id": "your_special_project",
  "private_key_id": "XXXXXXXXXXXXXXXXXXXXxx",
  "private_key": "-----BEGIN PRIVATE KEY-----\nNONONONO\n-----END PRIVATE KEY-----\n",
  "client_email": "somedude@your_special_project.iam.gserviceaccount.com",
  "client_id": "1234567890",
  "auth_uri": "https://accounts.google.com/o/oauth2/auth",
  "token_uri": "https://oauth2.googleapis.com/token",
  "auth_provider_x509_cert_url": "https://www.googleapis.com/oauth2/v1/certs",
  "client_x509_cert_url": "https://www.googleapis.com/robot/v1/metadata/x509/somedude%40your_special_project.iam.gserviceaccount.com"
}
```
As could be other gcr images which use simple basic auth

Comment (chen-keinan) [reply pkg/plugins/trivy/plugin.go:809]: just an Idea , make the validation that image is gcr and password is from type sa under the the following function:
`newSecretWithAggregateImagePullCredentials` and if so label the `secret` as  password-type : gcr-sa

**Review by chen-keinan — CHANGES_REQUESTED** [review]

**Comment by chen-keinan** [comment]
@btwseeu78 thank you for the contribution , I have added few comments

**Comment by chen-keinan** [comment]
@btwseeu78 Let me know if you need my help to promote your PR :)

**Comment by btwseeu78** [comment]
I'm currently on vacation if you can manage some time please take it forwarded else I will try as per review post vacation

**Comment by chen-keinan** [comment]
@btwseeu78 sure, enjoy your time off :)

**Comment by chen-keinan** [comment]
@btwseeu78 do you mind if I'll add some code changes to your PR ?

File: pkg/plugins/trivy/plugin.go [hunk 2]
```diff
@@ -41,6 +41,7 @@ const (
 )
 
 const (
+	GCPCR_Inage_Regex         = `^(gcr\.io.*|^([a-zA-Z0-9-]+)-*-*.docker.pkg.dev.*)`
```
Comment (chen-keinan) [inline pkg/plugins/trivy/plugin.go:44]: is this regex with suffix `docker.pkg.dev.*` will cover all gcr use case or is it specific ?

**Comment by chen-keinan** [comment]
@btwseeu78 I have push couple of small changes including docs update , please review docs update and make sure it make sense. I have added small comment regarding regex , please take a look and advice

File: pkg/plugins/trivy/plugin.go (hunk 2 above)
Comment (btwseeu78) [reply pkg/plugins/trivy/plugin.go:44]: please feel free to do any changes to you want. normally it should cover gcr or AR like europe-docker.pkg.dev

Comment (btwseeu78) [reply pkg/plugins/trivy/plugin.go:44]: @chen-keinan please feel free to add anything.

**Review by btwseeu78 — COMMENTED** [review]
fine by me

**Review by btwseeu78 — COMMENTED** [review]
no issues from my end. just this maybe would be better for creating secret.
---


kubectl create secret docker-registry gcr-json-key \
(out) --docker-server=eu.gcr.io \
(out) --docker-username=_json_key \
(out) --docker-password="$(cat ~/json-key-file.json)" \
(out) --docker-email=any@valid.email

**Comment by chen-keinan** [comment]
> ## no issues from my end. just this maybe would be better for creating secret.
> kubectl create secret docker-registry gcr-json-key (out) --docker-server=eu.gcr.io (out) --docker-username=_json_key (out) --docker-password="$(cat ~/json-key-file.json)" (out) --docker-email=[any@valid.email](mailto:any@valid.email)

sure , you can raise a PR to update docs

**Pull request merged**